src/store/contact-groups/index.ts

```diff
@@ -4,48 +4,119 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
+import { differenceBy, findIndex } from 'lodash';
 import { create } from 'zustand';
 
 import { ContactGroup } from '../../model/contact-group';
 
+function compareContactGroupName(nameA: string, nameB: string): number {
+	if (nameA > nameB) {
+		return 1;
+	}
+	if (nameB > nameA) {
+		return -1;
+	}
+	return 0;
+}
+
```

Comment:
This comparator is case sensitive, compare the lowercased strings to keep the same order the BE would apply